src/tides.f90

```diff
@@ -573,4 +561,336 @@ subroutine setTidesInitFlag(this)
 
   end subroutine setTidesInitFlag
 
+!=========================================================================
+!
+!> @anchor      getDelaunay_arg
+!!
+!> @brief       Compute the Delaunay arguments (IERS chapter 5)
+!> @author      Andrea Turchi (ATU)
+!!
+!> @date        <ul>
+!!                <li> 17.08.22 (initial design) </li>
+!!              </ul>
+!!
+!!-----------------------------------------------------------------------
+  subroutine get_Delaunay_arg(this, time_mjd, F_vect)
+
+    implicit none
+    class(Tides_class)                  :: this
+    real(dp), intent(in)                :: time_mjd  ! time in modified julian date
+    real(dp)                            :: time_jd   ! time in julian date
+    real(dp)                            :: jd_cent   ! time in julian centuries
+    real(dp)                            :: l         ! mean anomaly of the Moon
+    real(dp)                            :: l_prime   ! mean anomaly of the Sun
+    real(dp)                            :: F         ! mean longitude of the Moon minus Omega
+    real(dp)                            :: D         ! mean elongation of the Moon from the Sun
+    real(dp)                            :: Omega     ! mean longitude of the ascending node of the Moon
+    real(dp), dimension(5), intent(out) :: F_vect    ! Delaunay arguments vector
+
+    !get input time in Julian centuries
+    time_jd = time_mjd + jd245
+    jd_cent = (time_jd - jd2000)/julian_century
+    
+    !compute Delaunay arguments in degrees 
+    l = 134.96340251d0 + (1717915923.217800d0 * jd_cent + 31.879200d0 * jd_cent**2.d0 &
+        + 0.05163500d0 * jd_cent**3.d0 - 0.0002447000d0 * jd_cent**4.d0) * as2deg
+    l_prime = 357.52910918d0 + (129596581.048100d0 * jd_cent - 0.553200d0 * jd_cent**2.d0 &
+              + 0.00013600d0 * jd_cent**3.d0 - 0.0000114900d0 * jd_cent**4.d0) * as2deg
+    F = 93.27209062d0 + (1739527262.847800d0 * jd_cent - 12.751200d0 * jd_cent**2.d0 &
+        - 0.00103700d0 * jd_cent**3.d0 + 0.0000041700d0 * jd_cent**4.d0) * as2deg
+    D = 297.85019547d0 + (1602961601.209000d0 * jd_cent - 6.370600d0 * jd_cent**2.d0 &
+        + 0.00659300d0 * jd_cent**3.d0 - 0.0000316900d0 * jd_cent**4.d0) * as2deg
+    Omega = 125.04455501d0 + (- 6962890.543100d0 * jd_cent + 7.472200d0 * jd_cent**2.d0 &
+            + 0.00770200d0 * jd_cent**3.d0 - 0.0000593900d0 * jd_cent**4.d0) * as2deg
+
+    ! insert Delaunay arguments inside a vector
+    F_vect = (/l, l_prime, F, D, Omega/)                           
+
+  end subroutine get_Delaunay_arg
+
+!=========================================================================
+!
+!> @anchor      getDoodson_arg
+!!
+!> @brief       Compute the Doodson arguments (as Orekit)
```

Comment:
What does the Orekit reference mean here?

Reply:
It is just a way to remind me, or whoever is working on the code, where the equations come from. In chapter 6 of IERS report, multipliers are only inserted inside tables, therefore it would require a large matrix to store all of them. Instead, through the equations that I've taken from Orekit, it is possible to extract them from the Doodson number by considering one tide at a time.

Reply:
Can you then be more specific and point to where this can be found? Add a link to the source and point out that we used the approach.